lib/internal/streams/async_iterator.js

```diff
@@ -117,6 +127,11 @@ const ReadableStreamAsyncIteratorPrototype = ObjectSetPrototypeOf({
     return promise;
   },
 
+  // TODO(ronag): destroy streams on error?
+  // throw(err) {
+  //   destroy(stream, err);
+  // },
```

Comment:
@mcollina: Is this something we should have? i.e. currently I don't think streams are cleaned up if doing:

```js
try {
  for await (const chunk of readable) {
    throw new Error('asd');
  }
} catch (err) {
  assert.strictEqual(err.message. 'asd');
}
```

Reply:
Should probably be a separate PR... just noticed it while working on this